crates/resolver/src/name_server/name_server.rs

```diff
@@ -138,7 +138,7 @@ impl<P: ConnectionProvider> NameServerState<P> {
         // Unless the user specified that we should follow the configured order,
         // re-order the connections to prioritize UDP.
         if options.server_ordering_strategy != ServerOrderingStrategy::UserProvidedOrder {
-            connections.sort_by_key(|ns| (ns.protocol != Protocol::Udp) as u8);
```

Comment:
One interesting observation from https://github.com/hickory-dns/hickory-dns/pull/3264: I think (?) this code is _only_ exercised by unit tests.

When we construct the `NameServerState` for the root nameservers and nameservers for a specific zone, we hard-code the default `ServerOrderingStrategy`, `ServerOrderingStrategy::QueryStatistics`.

Reply:
Confusing (!), but I think you're right.